performance/plugins/extrinsicTransfer.ts

```diff
@@ -0,0 +1,113 @@
+import { Keyring } from "@polkadot/api";
+import BN from "bn.js";
+import { Mangata } from "mangata-sdk";
+import { testLog } from "../../utils/Logger";
+import { TestParams } from "../testParams";
+import { TestItem } from "./testItem";
+import { KeyringPair } from "@polkadot/keyring/types";
+import { UserFactory, Users } from "../../utils/Framework/User/UserFactory";
+import { Node } from "../../utils/Framework/Node/Node";
+import { MGA_ASSET_ID } from "../../utils/Constants";
+import { mintAsset } from "../../utils/tx";
+import { getApi, initApi } from "../../utils/api";
+import { preGenerateTransactions } from "./testRunner";
+
+function seedFromNum(seed: number): string {
+  return "//user//" + ("0000" + seed).slice(-4);
+}
+
+const mgaNodeandUsers = new Map<
+  number,
+  { mgaSdk: Mangata; users: { nonce: BN; keyPair: KeyringPair }[] }
+>();
+
+export class ExtrinsicTransfer implements TestItem {
+  async arrange(numberOfThreads: number, nodes: string[]): Promise<boolean> {
+    const keyring = new Keyring({ type: "sr25519" });
+    const mintPromises = [];
+    for (let nodeNumber = 0; nodeNumber < nodes.length; nodeNumber++) {
+      const node = nodes[nodeNumber];
+      const mga = await getMangata(node);
+      const mgaNode = new Node(node);
+      const sudo = UserFactory.createUser(Users.SudoUser, keyring, mgaNode);
+
+      const users: { nonce: BN; keyPair: KeyringPair }[] = [];
+      testLog.getLog().info("Fetching nonces for node " + nodeNumber);
+
+      //lets create as many of users as threads.
+      for (let i = 0; i < numberOfThreads; i++) {
+        const stringSeed = seedFromNum(i);
+        const keyPair = keyring.addFromUri(stringSeed);
+        const nonce = await mga.getNonce(keyPair.address);
+
+        //lets mint some MGA assets to pay fees
+        mintPromises.push(
+          mintAsset(
+            sudo.keyRingPair,
+            MGA_ASSET_ID,
+            keyPair.address,
+            new BN(10).pow(new BN(18))
+          )
+        );
+        users.push({ nonce: nonce, keyPair: keyPair });
+      }
+      mgaNodeandUsers.set(nodeNumber, { mgaSdk: mga, users: users });
+    }
+    const results = await Promise.all(mintPromises);
+    testLog
+      .getLog()
+      .info("All nonces and users fetched!" + JSON.stringify(results));
+
+    return true;
+  }
+  async act(testParams: TestParams): Promise<boolean> {
+    const preSetupThreads = await preGenerateTransactions(
+      testParams,
+      mgaNodeandUsers
+    );
+    const tx = await preSetupThreads[0][0][0];
+    await tx.send();
+    return true;
```

Comment:
@CodeByEddy  if you finish the governance and the multi validator stuff, you can take it from there!.
Ideally, we can create a method that uses the preSetupThreads, send the txs and run the threads based on the config.
Something similar to this loop should be enough I think.
https://github.com/paritytech/sub-flood/blob/master/src/index.ts#L147